src/api/app/services/workflows/yaml_downloader.rb

```diff
@@ -15,44 +15,48 @@ def call
     private
 
     def download_yaml_file
-      url = download_url
-      return download_gitlab_yaml_file if url.nil? && @scm_payload[:scm] == 'gitlab'
-
-      Down.download(url, max_size: MAX_FILE_SIZE)
-    rescue Down::Error => e
-      raise Token::Errors::NonExistentWorkflowsFile, "#{@token.workflow_configuration_url} could not be downloaded.\n#{e.message}" if @token.workflow_configuration_url.present?
-
-      # 'target_branch' can contain a commit sha (when tag push) instead of a branch name
-      raise Token::Errors::NonExistentWorkflowsFile, "#{@token.workflow_configuration_path} could not be downloaded from the SCM branch/commit #{@scm_payload[:target_branch]}." \
-                                                     "\nIs the configuration file in the expected place? Check #{DOCUMENTATION_LINK}\n#{e.message}"
+      # When an external URL is given, it prevails over the path.
+      if @scm_payload[:scm] == 'gitea' || @token.workflow_configuration_url.present?
+        download_gitea_or_configuration_url_yaml_file
+      elsif @scm_payload[:scm] == 'github'
+        download_github_yaml_file
+      elsif @scm_payload[:scm] == 'gitlab'
+        download_gitlab_yaml_file
+      end
     end
 
-    def download_url
-      # When an external URL is given, it prevails over the path.
-      return @token.workflow_configuration_url if @token.workflow_configuration_url.present?
+    def download_gitea_or_configuration_url_yaml_file
```

Comment:
I don't think it makes much sense to stuff this into one method just to `if` around in it again.

Reply:
Okay, moving it out of this PR and I'll separate them